package.json

```
@@ -50,32 +58,127 @@
         "path": "./syntaxes/markdown.tmLanguage.json"
       }
     ],
-    "themes": [
+    "commands": [
       {
-        "label": "D2 Light",
-        "uiTheme": "vs",
-        "path": "./themes/light-color-theme.json"
-      },
+        "command": "d2-viewer.ShowPreviewWindow",
+        "title": "Show d2 Preview Window",
+        "category": "d2"
+      }
+    ],
+    "keybindings": [
       {
-        "label": "D2 Dark",
-        "uiTheme": "vs-dark",
-        "path": "./themes/dark-color-theme.json"
+        "command": "d2-viewer.ShowPreviewWindow",
+        "key": "ctrl+shift+d",
+        "mac": "shift+cmd+d"
+      }
+    ],
+    "configuration": {
+      "type": "object",
+      "title": "d2 Viewer Configuration",
+      "properties": {
+        "d2-viewer.autoUpdate": {
+          "type": "boolean",
+          "default": true,
+          "description": "Automatically update the preview."
+        },
+        "d2-viewer.updateTimer": {
+          "type": "number",
+          "default": 1500,
+          "description": "Specifies the amount of time to wait between keystrokes before automatically updating the preview. (1000 = 1000ms = 1s)"
+        },
+        "d2-viewer.updateOnSave": {
+          "type": "boolean",
+          "default": false,
+          "description": "Automatically update on saving the d2 file.  Turn off to avoid excess updates when Auto Save is enabled."
```

```suggestion
          "description": "Automatically update on saving the d2 file.  Turn off to avoid excess updates when Auto Update is enabled."
```

I think we want to have the save event stop the refresh timer if this is on, so we won't need this second sentence and it will work well if both settings are enabled.

---

Fixed in PR #16